### DISCUSSION TITLE
Keeping my repo in sync with yours

### DISCUSSION THREAD

**Discussion by rymix** [DustinBrett/daedalOS#267] [General]
Hi Dustin, 

I’m interested in your thoughts regarding my little challenge. I cloned your repo back in December as a basis of some development I’m doing - specifically the creation of some new apps and some UI changes to your main project (amongst other things).

I guess this is an unofficial fork.

I’m keen to keep up with your continuous improvements, but as you can imagine, rebasing/merging onto an ever-divergent code base is a larger and larger challenge. I’ve done this twice and already I’m bored of it.

Can you think of a more efficient way for me to keep in sync with your changes?

**Comment by DustinBrett** [comment]
Hi Rymix (Steve?),

Cool that you've forked my repo. Sorry to hear you've had trouble with rebasing.

I would say it depends what changes you are doing but in general it is going to be a little tricky as I still change the code a lot on a daily basis. I would recommend you do changes in an isolated or modular fashion so that the main files stay as unchanged as possible, causing less conflicts.

In a typical OSS project I would imagine there would be some expectation of stability, but tbh I don't plan to ever stop changing this code and I want it to evolve along with technology and my understanding.

The more I know about the reasons/needs of forking, the better I can adjust my code in the future to be more modular and stable for specific customization purposes. I will look into the current deviations (https://github.com/rymix/steffm/commits?author=steveasmyle) to see if I can make the code easier to modify while allowing it to evolve.

Thanks for asking!
